### PR TITLE
Feature/user repository class

### PR DESCRIPTION
**Pull request by ASands17**
What does this PR do?
- Creates user and userRepository classes. 
- Creates user and userRepository test files.

Where should your reviewer start?
- User.js: line 1
- UserRepository.js: line 1
- User-test.js: line 1
- UserRepository-test.js: line 3

What (if any) features are you implementing?
- Creates classes and methods.

What (if anything) did you refactor?
- Refactors spacing

Were there any issues that arose?
- We resolved a merge conflict.

Is there anything that you need from your teammate?
- No.

Any other comments, questions, or concerns?
- No.

### CODE REVIEW

File: src/User.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    this.name = userDataObject.name;
+    this.address = userDataObject.address;
+    this.email = userDataObject.email;
+    this.strideLength = userDataObject.strideLength;
+    this.dailyStepGoal = userDataObject.dailyStepGoal;
+    this.friends = userDataObject.friends;
+  }
+
+  returnUserFirstName() {
+    return this.name.split(" ")[0];
```
Comment (irmakerdem) [inline src/User.js:13]: Whoa what a cool line of code that is 👀

File: src/UserRepository.js [hunk 2]
```diff
@@ -1,5 +1,28 @@
+import User from '../src/User';
+
 class UserRepository {
+  constructor(allUserData) {
+    this.data = allUserData;
+  }
+
+  getUserDataBasedOnId(id) {
```
Comment (irmakerdem) [inline src/UserRepository.js:8]: MERGE CONFLICT RESOLVED 🤡

File: src/UserRepository.js [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+    return user;
+  }
 
+  getAveStepGoalOfAllUsers() {
+    let totalStepGoal = this.data.reduce((totalSteps, currentUser) => {
+      totalSteps += currentUser.dailyStepGoal;
+      return totalSteps;
+    }, 0);
+    let averageStepGoal = totalStepGoal / this.data.length;
+    return Math.round(averageStepGoal);
```
Comment (catlynbowles) [inline src/UserRepository.js:24]: I love the use of Math.round() here. Good thinking! 🌺

File: test/UserRepository-test.js [hunk 4]
```diff
@@ -1,8 +1,97 @@
...
+      "strideLength": 4.3,
+      "dailyStepGoal": 10000,
+      "friends": [
+        16,
+        4,
+        8
+      ]
+      },
+      {
+        "id": 2,
```
Comment (irmakerdem) [inline test/UserRepository-test.js:28]: Need to change indentation

**Review by irmakerdem — APPROVED** [review]
Looks great! 😍

File: src/User.js (hunk 1 above)
Comment (catlynbowles) [inline src/User.js:13]: Great way to return first name. 🧛🏻‍♀️

**Review by catlynbowles — APPROVED** [review]

**Pull request merged**